### PR TITLE
Adding Spinner Properties

### PR DESCRIPTION
**Pull request by ReachAayush**
I have been working on issue #211. Also, to solve some issues with Spinner not having a default value as mentioned in issue #309, I set the initial Selection to the empty string and the initial SelectionIndex to 0. 

I have implemented successfully the following properties:
enabled
font bold/size/color
BeforeSelecting event and handler

I need help with the following properties:
background color -- currently I am able to change the background color of the selected item's TextView. This however does not change the color of the whole button and looks unpleasant. 

font italic/typeface -- Though these are currently implemented very similarly to Font Bold and TextColor, I cannot figure out why these methods are not working. 

text alignment -- There is a very strange bug happening with TextAlignment, if it is not set to the default value, nothing appears in the spinner. I cannot understand why this is happening and would appreciate help with this.

height -- I am still currently working on this.

### CODE REVIEW

File: appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java [hunk 1]
```diff
@@ -190,16 +471,24 @@ public void AfterSelecting(String selection){
   public void onItemSelected(AdapterView<?> parent, View view, int position, long id){
     //prevent AfterPicking triggering when component has just been instantiated.
     if (!isInitialized) {
-      isInitialized = true;
-      return;
+        isInitialized = true;
```
Comment (josmas) [inline appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java]: indentation?

File: appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java [hunk 2]
```diff
@@ -190,16 +471,24 @@ public void AfterSelecting(String selection){
...
+      BackgroundColor(backgroundColor);
+      TextColor(textColor);
+      TextAlignment(textAlignment);
   }
 
   public void onNothingSelected(AdapterView<?> parent){
     view.setSelection(0);
   }
 
+
```
Comment (josmas) [inline appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java]: extra line?

File: appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java [hunk 3]
```diff
@@ -55,16 +66,274 @@ public Spinner(ComponentContainer container) {
...
+    TextViewUtil.setBackgroundColor(selectedTextView, argb);
+  }
+
   @Override
   public View getView(){
     return view;
   }
 
   /**
+   * Returns true if the Spinner is active and clickable.
```
Comment (josmas) [inline appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java]: Are most of these functions very similar to functions in Button or ButtonBase? Any chance to extract them out to a collaboration class and use them in all classes needed?

File: appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java [hunk 4]
```diff
@@ -55,16 +66,274 @@ public Spinner(ComponentContainer container) {
     adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
     view.setAdapter(adapter);
     view.setOnItemSelectedListener(this);
+    //Aayush
+    Enabled(true);
+      FontBold(bold);
```
Comment (josmas) [inline appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java]: indentation?

File: appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java [hunk 5]
```diff
@@ -55,16 +66,274 @@ public Spinner(ComponentContainer container) {
     adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
     view.setAdapter(adapter);
     view.setOnItemSelectedListener(this);
+    //Aayush
```
Comment (josmas) [inline appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java]: delete comment with name

**Comment by josmas** [comment]
Hi @ReachAayush thanks for the pull request. I see it's still a work in progress but can you please have a look a the comments inline?
Also, if you need help with something, please also open a thread in the oss forum explaining what's needed. Might get some readers there that won't notice a pull request.

File: appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java (hunk 1 above)
Comment (colinwirt) [inline appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java]: Line 400 in the Elements property setter causes the first user-initiated selection from calling AfterSelection:
     //avoid firing off the OnItemSelect when component is initialized or data is changed
     isInitialized = false;

This is broken badly and basically makes the Elements method unusable!

This is not a problem for ElementsFromString.

**Comment by josmas** [comment]
this pull request needs answers to comments. @ReachAayush are you planning on finishing this up or shall I close it?

**Comment by josmas** [comment]
no reply in a month, closing down.

**Pull request closed without merging**